### PR TITLE
Make sure the viewport is never of size 0

### LINKED ISSUE #1003: Rendering into 1D texture uses viewport of heigth 1
reported by @Bastacyclop 
2d2826e4a40448ad8f480d5ae1e2a5e2

### PR DESCRIPTION
**Pull request by Bastacyclop**
Fixes #1003

### CODE REVIEW

File: src/core/src/pso.rs [hunk 1]
```diff
@@ -258,27 +259,36 @@ impl<R: Resources> PixelTargetSet<R> {
...
             self.stencil = Some(view.clone());
         }
-        self.size = max(self.size, dim);
+        self.set_size(dim);
+    }
+
+    fn set_size(&mut self, dim: texture::Dimensions) {
+        debug_assert!(!self.sized || self.size == dim);
+        self.sized = true;
+        self.size = dim;
```
Comment (kvark) [inline src/core/src/pso.rs]: let's use `Option<>` instead?

Comment (Bastacyclop) [inline src/core/src/pso.rs]: should the backend unwrap it or handle the case where there is no pixel target differently ?

Comment (kvark) [inline src/core/src/pso.rs]: perhaps, use `unwrap_or` at least, may be changed later to some error if we want

File: src/core/src/pso.rs [hunk 2]
```diff
@@ -257,28 +257,42 @@ impl<R: Resources> PixelTargetSet<R> {
...
+        self.set_dimensions(dim);
+    }
+
+    fn set_dimensions(&mut self, dim: texture::Dimensions) {
+        debug_assert!(self.dimensions.map(|d| d == dim).unwrap_or(true));
+        self.dimensions = Some(dim);
+    }
+
+    /// Get the rendering dimensions or empty ones
+    pub fn unwrap_dimensions(&self) -> texture::Dimensions {
```
Comment (kvark) [inline src/core/src/pso.rs]: Might as well call this `get_viewport` and put a `max` in there ;)

**Comment by kvark** [comment]
Looks good, thanks! Waiting for AppVeyor now.

File: src/backend/vulkan/src/command.rs [hunk 3]
```diff
@@ -146,11 +146,12 @@ impl command::Buffer<Resources> for Buffer {
 
     fn bind_pixel_targets(&mut self, pts: pso::PixelTargetSet<Resources>) {
         let (dev, vk) = self.share.get_device();
+        let view = pts.get_view();
         let vp = vk::Viewport {
             x: 0.0,
             y: 0.0,
-            width: pts.size.0 as f32,
-            height: pts.size.1 as f32,
+            width: max(view.0, 1) as f32,
```
Comment (kvark) [inline src/backend/vulkan/src/command.rs]: these `max` should not be needed

**Comment by kvark** [comment]
@homu r+

**Pull request merged**